src/main/java/com/replaymod/render/rendering/VideoRenderer.java

```diff
@@ -631,19 +638,40 @@ public boolean drawGui() {
     }
 
     private boolean displaySizeChanged() {
+        int realWidth = mc.getWindow().getWidth();
+        int realHeight = mc.getWindow().getHeight();
+        return displayWidth != realWidth || displayHeight != realHeight;
+    }
+
+    private boolean framebufferSizeChanged() {
         int realWidth = mc.getWindow().getFramebufferWidth();
         int realHeight = mc.getWindow().getFramebufferHeight();
         if (realWidth == 0 || realHeight == 0) {
             // These can be zero on Windows if minimized.
             // Creating zero-sized framebuffers however will throw an error, so we never want to switch to zero values.
             return false;
         }
-        return displayWidth != realWidth || displayHeight != realHeight;
+        return framebufferWidth != realWidth || framebufferHeight != realHeight;
     }
 
     private void updateDisplaySize() {
-        displayWidth = mc.getWindow().getFramebufferWidth();
-        displayHeight = mc.getWindow().getFramebufferHeight();
+        displayWidth = mc.getWindow().getWidth();
+        displayHeight = mc.getWindow().getHeight();
+    }
+
+    private void updateFramebufferSize() {
+        framebufferWidth = mc.getWindow().getFramebufferWidth();
+        framebufferHeight = mc.getWindow().getFramebufferHeight();
+    }
+
+    private void recalculateFramebufferSize() {
```

Comment:
Note: This may be replaced by a call to `window.updateFramebufferSize()` but it is private and I do not want to touch ReplayMod accesswideners.

Reply:
RM doesn't use AWs because those do not port to 1.12.2 and below (and AWs are in general not recommended unless Mixin cannot yet do what you need). Instead, you can declare an `invokeUpdateFramebufferSize` `@Invoker` in `MainWindowAccessor`.
That also happens to make backporting easier (`Window` was only added in 1.14), you just have to implement the method in https://github.com/ReplayMod/ReplayMod/blob/develop/versions/1.12.2/src/main/java/com/replaymod/core/versions/Window.java. Old MC versions don't differentiate between window and framebuffer size, so it'd probably just be a noop.

Reply:
I can't seem to understand how to backport properly. I implemented the method in the 1.12 overwrite file and got an error 

> /Users/andrews/Desktop/ReplayMod/versions/1.14.4/build/preprocessed/main/java/com/replaymod/render/mixin/MainWindowAccessor.java:18: error: Could not locate @Invoker target updateFramebufferSize in target net.minecraft.client.util.Window
    @Invoker

Reply:
Looks like that method didn't have a name in 1.14.4 (and there's a bug in the preprocessor where it doesn't remap to intermediary names). Try adding
```
net.minecraft.client.util.Window updateFramebufferSize() method_4483()
```
to `mapping-fabric-1.15.2-1.14.4.txt`.

I also just noticed that there's a 1.12.2 overwrite file for `MainWindowAccessor` as well (afaict there shouldn't need to be one but that's out of scope for this PR), so you'll have to declare your invoker method in there as well, just don't add the actual `@Invoker`. Though I'm not sure if mixin allows that. If it doesn't, then you'll also have to remove the class from the mixin config for 1.12.2 and below (just move its entry up one line, inside the preprocessor block); it's only being used as a plain old interface anyway.

Reply:
I still got the same error after I did this.

Reply:
Eh, then don't worry about it, I'll deal with it when I merge it.
Should be good as is for now.